crates/ra_hir/src/code_model.rs

```diff
@@ -275,19 +286,9 @@ impl Module {
     pub fn find_use_path(
         self,
         db: &dyn HirDatabase,
-        item: ModuleDef,
+        item: ItemInNs,
     ) -> Option<hir_def::path::ModPath> {
-        // FIXME expose namespace choice
-        hir_def::find_path::find_path(db.upcast(), determine_item_namespace(item), self.into())
-    }
-}
-
-fn determine_item_namespace(module_def: ModuleDef) -> ItemInNs {
-    match module_def {
-        ModuleDef::Static(_) | ModuleDef::Const(_) | ModuleDef::Function(_) => {
-            ItemInNs::Values(module_def.into())
-        }
-        _ => ItemInNs::Types(module_def.into()),
+        hir_def::find_path::find_path(db.upcast(), item, self.into())
```

Comment:
Hm, look strange that we do an upcast right away here. It looks like we should accept `DefDatabase` in the first place or not have this method at all (i.e. just use `ra_hir_def::find_path`) on the call site instead since this method ends up to be a *very* thin wrapper?

Reply:
I've removed an upcast, but it's all around in the other parts in the file, so maybe that is indended and @matklad had had a reson to do so instead of replacing the DB types himself.

The thin wrapper forms a public API and this way it looks way more convenient than a standalone method to me.